wcfsetup/install/files/lib/system/worker/UserContentRemoveWorker.class.php

```diff
@@ -126,32 +135,36 @@ private function generateData() {
 			}
 		}
 		
+		$contentProviders = ObjectTypeCache::getInstance()->getObjectTypes('com.woltlab.wcf.content.userContentProvider');
```

Comment:
This could be removed if `$this->contentProvider` would store the `$objectType` instead of the `$objectTypeName`. Then you could directly sort the actual list and could get rid of the condition within the loop.

Reply:
Keep in mind, that the class variable `$contentProvider` is a parameter, which is read in `validate()`. In the end, it doesn't matter where the whole thing is sorted (and does not clean up the code).

Reply:
I was thinking about this (untested):

```
diff --git i/wcfsetup/install/files/lib/system/worker/UserContentRemoveWorker.class.php w/wcfsetup/install/files/lib/system/worker/UserContentRemoveWorker.class.php
index 1649009ed5..795cae338e 100644
--- i/wcfsetup/install/files/lib/system/worker/UserContentRemoveWorker.class.php
+++ w/wcfsetup/install/files/lib/system/worker/UserContentRemoveWorker.class.php
@@ -1,5 +1,6 @@
 <?php
 namespace wcf\system\worker;
+use wcf\data\object\type\ObjectType;
 use wcf\data\object\type\ObjectTypeCache;
 use wcf\data\user\User;
 use wcf\data\user\UserList;
@@ -113,6 +114,9 @@ class UserContentRemoveWorker extends AbstractWorker {
 			'count' => 0
 		];
 		
+		/** @var ObjectType[] $contentProviders */
+		$contentProviders = [];
+		
 		// add the required object types for the select content provider
 		if (is_array($this->contentProviders)) {
 			foreach ($this->contentProviders as $contentProvider) {
@@ -128,13 +132,14 @@ class UserContentRemoveWorker extends AbstractWorker {
 							throw new \RuntimeException('Unknown required object type "' . $objectTypeName . '" for object type "' . $contentProvider . '" given.');
 						}
 						
-						$this->contentProviders[] = $objectTypeName;
+						$contentProviders[] = $objectType;
 					}
 				}
 			}
 		}
-		
-		$contentProviders = ObjectTypeCache::getInstance()->getObjectTypes('com.woltlab.wcf.content.userContentProvider');
+		else {
+			$contentProviders = ObjectTypeCache::getInstance()->getObjectTypes('com.woltlab.wcf.content.userContentProvider');
+		}
 		
 		// sort object types
 		uasort($contentProviders, function ($a, $b) {
@@ -145,22 +150,20 @@ class UserContentRemoveWorker extends AbstractWorker {
 		});
 		
 		foreach ($contentProviders as $contentProvider) {
-			if ($this->contentProviders === null || (is_array($this->contentProviders) && in_array($contentProvider->objectType, $this->contentProviders))) {
-				foreach ($this->users as $user) {
-					/** @var IUserContentProvider $processor */
-					$processor = $contentProvider->getProcessor();
-					$contentList = $processor->getContentListForUser($user);
-					$count = $contentList->countObjects();
+			foreach ($this->users as $user) {
+				/** @var IUserContentProvider $processor */
+				$processor = $contentProvider->getProcessor();
+				$contentList = $processor->getContentListForUser($user);
+				$count = $contentList->countObjects();
+				
+				if ($count) {
+					$this->data['provider'][] = [
+						'userID' => $user->userID,
+						'objectTypeID' => $contentProvider->objectTypeID,
+						'count' => $count
+					];
 					
-					if ($count) {
-						$this->data['provider'][] = [
-							'userID' => $user->userID,
-							'objectTypeID' => $contentProvider->objectTypeID,
-							'count' => $count
-						];
-						
-						$this->data['count'] += ceil($count / $this->limit) * $this->limit;
-					}
+					$this->data['count'] += ceil($count / $this->limit) * $this->limit;
 				}
 			}
 		}
```

With Whitespace ignored:

```
diff --git i/wcfsetup/install/files/lib/system/worker/UserContentRemoveWorker.class.php w/wcfsetup/install/files/lib/system/worker/UserContentRemoveWorker.class.php
index 1649009ed5..795cae338e 100644
--- i/wcfsetup/install/files/lib/system/worker/UserContentRemoveWorker.class.php
+++ w/wcfsetup/install/files/lib/system/worker/UserContentRemoveWorker.class.php
@@ -1,5 +1,6 @@
 <?php
 namespace wcf\system\worker;
+use wcf\data\object\type\ObjectType;
 use wcf\data\object\type\ObjectTypeCache;
 use wcf\data\user\User;
 use wcf\data\user\UserList;
@@ -113,6 +114,9 @@ class UserContentRemoveWorker extends AbstractWorker {
 			'count' => 0
 		];
 		
+		/** @var ObjectType[] $contentProviders */
+		$contentProviders = [];
+		
 		// add the required object types for the select content provider
 		if (is_array($this->contentProviders)) {
 			foreach ($this->contentProviders as $contentProvider) {
@@ -128,13 +132,14 @@ class UserContentRemoveWorker extends AbstractWorker {
 							throw new \RuntimeException('Unknown required object type "' . $objectTypeName . '" for object type "' . $contentProvider . '" given.');
 						}
 						
-						$this->contentProviders[] = $objectTypeName;
+						$contentProviders[] = $objectType;
 					}
 				}
 			}
 		}
-		
+		else {
 			$contentProviders = ObjectTypeCache::getInstance()->getObjectTypes('com.woltlab.wcf.content.userContentProvider');
+		}
 		
 		// sort object types
 		uasort($contentProviders, function ($a, $b) {
@@ -145,7 +150,6 @@ class UserContentRemoveWorker extends AbstractWorker {
 		});
 		
 		foreach ($contentProviders as $contentProvider) {
-			if ($this->contentProviders === null || (is_array($this->contentProviders) && in_array($contentProvider->objectType, $this->contentProviders))) {
 			foreach ($this->users as $user) {
 				/** @var IUserContentProvider $processor */
 				$processor = $contentProvider->getProcessor();
@@ -164,7 +168,6 @@ class UserContentRemoveWorker extends AbstractWorker {
 			}
 		}
 	}
-	}
 	
 	/**
 	 * @inheritDoc
```

Which I'd say *is* cleaner, because the `if` could be removed.